graphql-java-support/src/main/java/com/apollographql/federation/graphqljava/Federation.java

```diff
@@ -184,4 +201,25 @@ private static RuntimeWiring ensureFederationDirectiveDefinitionsExist(
 
     return builder;
   }
+
+  public static boolean isFederation2(TypeDefinitionRegistry typeDefinitionRegistry) {
+    return typeDefinitionRegistry.getSchemaExtensionDefinitions().stream()
+        .anyMatch(
+            schemaExtensionDefinition ->
+                schemaExtensionDefinition.getDirectives().stream()
+                    .anyMatch(
+                        directive ->
+                            directive.getName().equals("link")
+                                && directive.getArguments().stream()
+                                    .anyMatch(
+                                        argument -> {
+                                          Value value = argument.getValue();
+                                          return argument.getName().equals("url")
+                                              && value instanceof StringValue
+                                              && ((StringValue) value)
+                                                  .getValue()
+                                                  .equals(
+                                                      "https://specs.apollo.dev/federation/v2.0");
+                                        })));
```

Comment:
Yikes that's a nice pyramid of doom 🙈 . If there's a way to improve this, I'll take it!

Reply:
To be fair I think the "pyramid" effect is due to the formatter 😁. You could maybe extract a few bits in dedicated methods but to me it's readable enough as-is.

Reply:
Yes, exactly! I had it with a lot less levels of indentation before I ran `./mvnw spotless:apply` 😬